docs/crefine-notes.md

```diff
@@ -243,14 +243,49 @@ However here are some tricks when it could go wrong:
   just before
 
 
-* sometimes the precondition generated by `wp` or `vcg` contains False somewhere (we figure it out while trying to prove the 'last big subgoal')
-so a simplification before the `wp`/`vcg` is needed...
+* sometimes the precondition generated by `wp` or `vcg` contains `False`
+  somewhere (we figure it out while trying to prove the 'last big subgoal') so a
+  modification before `vcg` is needed. See the next section for debugging tips
```

Comment:
Very minor, but why is it just `vcg` in this line now and not `wp`/`vcg`?

Reply:
`wp` no longer produces just `False`, but would fail to apply if the precondition doesn't depend on the right parameters. You can still get funky goals if your assumptions have a schematic in them, but then the proof is already broken in other ways.

Reply:
I've not really seen this case happen in practice. What I do see is that the schematic precondition doesn't work right, the simp outside/inside vcg resolves it to False, and we get a postcondition that has subterms like: `ALL t. some_var' t = foo --> False --> P (some_var t) /\ ...`  which then when hit by the next simp become `ALL t. some_var' t = foo`. Since `vcg` uses simp internally these are very difficult to spot. Please see my overall comment on this PR below.